### DISCUSSION TITLE
Bad appearance of road

### DISCUSSION THREAD

**Discussion by devolod** [maplibre/maplibre-gl-js#3858] [Q&A] [answered]
Hi All!

I try to display road as line with borders. Source is vector tiles. Layers are  two styles one above other like this:
road-border
`{'type': 'line', 'paint': { 'line-color':  '#555',  'line-width' : 7}}`, 
road-fill
`{'type': 'line', paint': { 'line-color': '#fff', 'line-width' : 6}}, `

The result is unsmooth border:    
![image](https://github.com/maplibre/maplibre-gl-js/assets/11441427/be3eafd2-3fa2-401a-98b7-2727e48c32f2)
Especially, it is happens when `pitch > 0`

Is there way to improve the appearance of road?

**Comment by wipfli** [comment]
Maybe increase the casing line width. That will make the casing look stronger, to compensate, you can change the casing color to be a bit more white...

**Comment by HarelM** [comment, marked as answer]
A single pixel line can get that way when tilted or rotated, I would recommend increasing the width.

**Comment by devolod** [comment]
Thanks for answers, guys!  I meant some `antialiasing` is switched on by default. By the way, what is best practice to display the roads?
